### ISSUE TITLE
Error of "Received fragmented control frame"

### ISSUE THREAD

**Issue by nttstar** [fafhrd91/aiohttp#7696]
### Describe the bug

When I'm using discord.py(which heavily relies on aiohttp), after running for a period of time, there is a certain probability that this error will be triggered. But when I searched for this error across the entire internet, I didn't find any information. I want to know whether this is a bug, or caused by some specific hardware conditions? Is it possible to resolve or ignore this issue at the aiohttp library level?

Full stack:

```
File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 860, in run
    asyncio.run(runner())
  File "/opt/conda/envs/pytorch/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/conda/envs/pytorch/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 849, in runner
    await self.start(token, reconnect=reconnect)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 778, in start
    await self.connect(reconnect=reconnect)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/shard.py", line 468, in connect
    raise item.error
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/shard.py", line 164, in worker
    await self.ws.poll_event()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/gateway.py", line 626, in poll_event
    raise msg.data
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/client_ws.py", line 196, in close
    msg = await self._reader.read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 657, in read
    return await super().read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 627, in read
    raise self._exception
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/client_ws.py", line 229, in receive
    msg = await self._reader.read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 657, in read
    return await super().read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 288, in feed_data
    return self._feed_data(data)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 295, in _feed_data
    for fin, opcode, payload, compressed in self.parse_frame(data):
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 459, in parse_frame
    raise WebSocketError(
aiohttp.http_websocket.WebSocketError: Received fragmented control frame
```

Many thanks for any helpful information.

ref: https://github.com/Rapptz/discord.py/discussions/9561

### To Reproduce

Hard to reproduce, randomly triggered.

### Expected behavior

No error occurred, or the error can be ignored.

### Logs/tracebacks

```python-traceback
File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 860, in run
    asyncio.run(runner())
  File "/opt/conda/envs/pytorch/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/conda/envs/pytorch/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 849, in runner
    await self.start(token, reconnect=reconnect)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/client.py", line 778, in start
    await self.connect(reconnect=reconnect)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/shard.py", line 468, in connect
    raise item.error
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/shard.py", line 164, in worker
    await self.ws.poll_event()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/discord/gateway.py", line 626, in poll_event
    raise msg.data
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/client_ws.py", line 196, in close
    msg = await self._reader.read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 657, in read
    return await super().read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 627, in read
    raise self._exception
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/client_ws.py", line 229, in receive
    msg = await self._reader.read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 657, in read
    return await super().read()
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 288, in feed_data
    return self._feed_data(data)
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 295, in _feed_data
    for fin, opcode, payload, compressed in self.parse_frame(data):
  File "/opt/conda/envs/pytorch/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 459, in parse_frame
    raise WebSocketError(
aiohttp.http_websocket.WebSocketError: Received fragmented control frame
```
```


### Python Version

```console
$ python --version
3.9.13
```

### aiohttp Version

```console
$ python -m pip show aiohttp
```

### multidict Version

```console
$ python -m pip show multidict
Name: aiohttp
Version: 3.8.3
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Location: /opt/conda/envs/pytorch/lib/python3.9/site-packages
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by: discord.py, gradio
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.8.1
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /opt/conda/envs/pytorch/lib/python3.9/site-packages
Requires: idna, multidict
Required-by: aiohttp
```

### OS

Ubuntu-20.04

### Related component

Server

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by Dreamsorcerer** [comment]
Looks like the other end of the connection is sending invalid websocket messages. If you can hook into that part of the discord library, then I'd assume you can catch the error. Not sure what you want to do upon receiving such an exception though, maybe just close the connection...

**Comment by nttstar** [comment]
Hi @Dreamsorcerer thanks for your reply. How about just try-catching and **ignoring** such type of error web-socket frames in my application? Is it the common practice?

**Comment by Dreamsorcerer** [comment]
Well, no, you've already stated that you couldn't find similar issues when searching, so I'd probably guess it is a rather unusual server that has messed up the websocket protocol?

**Comment by nttstar** [comment]
Hello, what I mean is at the application level, if such error messages are ignored, would it generally not cause problems?

**Comment by Dreamsorcerer** [comment]
No idea, I've never seen the issue before.

**Comment by weiss-gal** [comment]
I got the same problem on my discord bot, 

here is my call stack:
```plaintext
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/chunka_bank/production/cb_bot/cb_bot.py", line 187, in <module>
    main(sys.argv)
  File "/home/chunka_bank/production/cb_bot/cb_bot.py", line 184, in main
    client.run(bot_token)
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/discord/client.py", line 860, in run
    asyncio.run(runner())
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/discord/client.py", line 849, in runner
    await self.start(token, reconnect=reconnect)
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/discord/client.py", line 778, in start
    await self.connect(reconnect=reconnect)
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/discord/client.py", line 659, in connect
    await self.ws.poll_event()
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/discord/gateway.py", line 626, in poll_event
    raise msg.data
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/client_ws.py", line 244, in receive
    msg = await self._reader.read()
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/streams.py", line 663, in read
    return await super().read()
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/streams.py", line 622, in read
    await self._waiter
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/http_websocket.py", line 314, in feed_data
    return self._feed_data(data)
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/http_websocket.py", line 321, in _feed_data
    for fin, opcode, payload, compressed in self.parse_frame(data):
  File "/home/chunka_bank/.local/lib/python3.8/site-packages/aiohttp/http_websocket.py", line 486, in parse_frame
    raise WebSocketError(
aiohttp.http_websocket.WebSocketError: Received fragmented control frame


```

I'm using python version 3.8.10

aiohttp version:
```plaintext
$ python3 -m pip show aiohttp
Name: aiohttp
Version: 3.9.1
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: None
Author-email: None
License: Apache 2
Location: /home/chunka_bank/.local/lib/python3.8/site-packages
Requires: aiosignal, attrs, async-timeout, multidict, yarl, frozenlist
Required-by: discord.py
```

ubuntu version:
```plaintext
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal

```


It only happened to me once on a bot agent running for several weeks, 
let me know if there is any more meaningfull data I can extract

**Comment by weiss-gal** [comment]
Also, @Dreamsorcerer , 
maybe it makes sense to loop in someone from the discord team, 
as the remote peer of the websocket is probably at their side.

**Comment by weiss-gal** [comment]
From further looking, 
this seems to be the same as the problem reported here:
https://github.com/Rapptz/discord.py/discussions/9561

which is expected to be fixed in version 2.4 of the discord python library:
https://github.com/Rapptz/discord.py/commit/163a86c4a031f7ecd856f45a74da47e63b17bb16

so I believe this ticket can be closed unless you see anything particular that should be handled in aio library.

**Comment by Dreamsorcerer** [comment]
Would probably be good if someone can confirm the actual details of the message, but it can probably be closed unless someone confirms an issue on our side. Currently it still seems like discord is sending the wrong data to the websocket. The "fix" appears to be treating the error as a close event and therefore closing the socket gracefully (and maybe reconnecting).

**Comment by llimonix** [comment]
> From further looking, this seems to be the same as the problem reported here: [Rapptz/discord.py#9561](https://github.com/Rapptz/discord.py/discussions/9561)
> 
> which is expected to be fixed in version 2.4 of the discord python library: [Rapptz/discord.py@163a86c](https://github.com/Rapptz/discord.py/commit/163a86c4a031f7ecd856f45a74da47e63b17bb16)
> 
> so I believe this ticket can be closed unless you see anything particular that should be handled in aio library.

https://github.com/Rapptz/discord.py/commit/163a86c4a031f7ecd856f45a74da47e63b17bb16

I went into the file and changed these lines and now my bot does not crash. A simple solution to this problem
